js/cardview.js

```diff
@@ -16,28 +16,118 @@ let Card = React.createClass({
+    let attackingCard;
+    for (let card of window.game.current().board) {
+      if (card.hasFocus) {
+        attackingCard = card;
+        hasActiveCard = true;
+        break;
+      }
+      fromAttackIndex++;
+    }
+    let toIndex = window.game.opponent().board.indexOf(this.props.cardInfo);
+    if (toIndex != -1 && hasActiveCard) {
+      this.attack(fromAttackIndex, toIndex, attackingCard);
+      return;      
+    }
+
+    // PLAY CARD FROM HAND
+    // can click to play cards in hand
+    let fromIndex = window.game.current().hand.indexOf(this.props.cardInfo);
+    if (fromIndex != -1) {
+      this.playFromHand(fromIndex, this.props.cardInfo);
       return;
     }
     
-    this.props.hasFocus = !this.props.hasFocus;
-    // play card on 2nd click
-    if (!this.props.hasFocus) {
-      var fromIndex = window.game.current().hand.indexOf(this.props.cardInfo);
-      console.log(fromIndex);
-      console.log(window.game.current().hand);
-      var move = {"op":"play", 
+    // HIGHLIGHT CARD, OR SMASH FACE IF ALREADY HIIGHLIGHTED
+    // can click a card in play if it's in the player's board
+    // and it's not summoning sick or already attacked
+    fromIndex = window.game.current().board.indexOf(this.props.cardInfo);
+    if (fromIndex != -1 && 
+        !this.props.cardInfo.enteredPlayThisTurn &&
+        !this.props.cardInfo.hasAttacked) {
+      this.clickCardInPlay(fromIndex, this.props.cardInfo);
+      return;
+    }
+
+    // if it's not in their hand or board, then do nothing
+
+  },
+  
+  // smash creature's face
+  attackCreature: function(fromAttackIndex, toIndex, attackingCard) {
+    let move = {"op":"attack", 
+                "from":fromAttackIndex,
+                "to": toIndex
+               };
+    window.client.makeLocalMove(move);
+    attackingCard.hasAttacked = true;
+  },
+
+  // highlight a card when clicked, play when double clicked
+  playFromHand: function(fromIndex, card) {
+    // card can only be highlighted and played if player has enough mana
+    if (this.props.cardInfo.cost < this.props.player.mana) {
+      let moveClosure = function() {
+        let move = {"op":"play", 
+                    "from":fromIndex
+                   };
+        window.client.makeLocalMove(move);
+        card.enteredPlayThisTurn = true;
+      }
+      this.highlightOrPlayMove(moveClosure);    
+    }
+  },
+
+  // highlight a card when clicked, attack face when double clicked
+  clickCardInPlay: function(fromIndex, card) {
+    let moveClosure = function() {
+      let move = {"op":"face", 
                   "from":fromIndex
                  };
       window.client.makeLocalMove(move);
-    } else {
-      this.forceUpdate();
+      card.hasAttacked = true;      
     }
+    this.highlightOrPlayMove(moveClosure);
   },
 
+  // when a card is clicked, highlight it, play it, or attack with it
+  highlightOrPlayMove: function(moveClosure) {
+    this.props.cardInfo.hasFocus = !this.props.cardInfo.hasFocus;
```

Comment:
Hmm something is unreacty here. I think it's related to why you need to call forceUpdate. I think you are not supposed to modify props. Instead, if you're going to be modifying something, put it on state.